### DISCUSSION TITLE
Upload media without sending it

### DISCUSSION THREAD

**Discussion by Nikeware** [reo7sp/tgbot-cpp#205] [Q&A] [answered]
Hello all,

How to with tgbot-cpp  library upload media (photo or video) without actually sending it to the chat? I see that core API from Telegram itself have some appropriate method "[uploadMedia](https://core.telegram.org/method/messages.uploadMedia)", but I don't found it in tgbot-cpp.

Thanks!

**Comment by reo7sp** [comment, marked as answer]
`messages.uploadMedia` belongs to the MTProto client API, not the HTTP Bot API wrapped by tgbot-cpp. Bot API files are uploaded as part of a send/edit operation; there is no standalone upload-only method. Cache the returned `file_id` for reuse.
